File: app/src/main/java/com/example/webfactory/ui/login/LoginFragment.java
package com.example.webfactory.ui.login;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import com.example.webfactory.MainActivity;
import com.example.webfactory.R;
import com.example.webfactory.databinding.FragmentLoginBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.snackbar.Snackbar;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import java.util.Objects;

public class LoginFragment extends Fragment {
    private FragmentLoginBinding binding;
    private FirebaseAuth mAuth;
    private FirebaseAuth.AuthStateListener mAuthListener;
    private NavController navController;

    public LoginFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
       return inflater.inflate(R.layout.fragment_login, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding = FragmentLoginBinding.bind(view);
        navController = Navigation.findNavController(view);
        mAuth = FirebaseAuth.getInstance();

        binding.login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                login(binding.username.getText().toString(), binding.password.getText().toString());
            }
        });

        binding.register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                register(binding.username.getText().toString(), binding.password.getText().toString());
            }
        });

        FirebaseUser user = mAuth.getCurrentUser();
        if (user != null) {
//            Intent intent = new Intent(getContext(), MainActivity.class);
//            startActivity(intent);
            //navController.navigate(R.id.action_loginFragment_to_nav_home);
        }

        mAuthListener = new FirebaseAuth.AuthStateListener() {
            @Override
            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                FirebaseUser user = firebaseAuth.getCurrentUser();
                if (user != null) {
                    //navController.navigate(R.id.action_loginFragment_to_nav_home);
                } else {

                }
            }
        };
    }

    private void login(String email, String password) {
        mAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener((Activity) getContext(), new OnCompleteListener<AuthResult>() {
            @Override
            public void onComplete(@NonNull Task<AuthResult> task) {
                if (task.isSuccessful()) {
                    Toast.makeText(getContext(), "Авторизация прошла успешно!", Toast.LENGTH_SHORT).show();
                    navController.navigate(R.id.action_loginFragment_to_nav_home);
                } else {
                    Toast.makeText(getContext(), "Авторизация провалена!", Toast.LENGTH_SHORT).show();

                }
            }
        });
    }

        private void register(String email, String password){
            mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener((Activity) getContext(), new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {
                    if (task.isSuccessful()){
                        Toast.makeText(getContext(), "Регистрация прошла успешно!", Toast.LENGTH_SHORT).show();
                    }else{
                        Toast.makeText(getContext(), "Регистрация провалена!", Toast.LENGTH_SHORT).show();

                    }
                }
            });
        }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
    }

File: app/src/main/java/com/example/webfactory/presentation/login/LoginFragment.kt
package com.example.webfactory.presentation.login

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.asLiveData
import androidx.navigation.NavController
import androidx.navigation.Navigation.findNavController
import com.example.domain.models.User
import com.example.webfactory.R
import com.example.webfactory.databinding.FragmentLoginBinding
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch

class LoginFragment : Fragment() {

    private var _binding: FragmentLoginBinding? = null
    private val binding get() = _binding!!
    private lateinit var navController: NavController

    private lateinit var viewModel: LoginFragmentViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_login, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        _binding = FragmentLoginBinding.bind(view)
        navController = findNavController(view)

        viewModel = ViewModelProvider(
            this,
            LoginFragmentViewModelFactory(requireContext())
        )
            .get(LoginFragmentViewModel::class.java)

        isUserAuth()

        binding.login.setOnClickListener {
            login(
                User(email = binding.username.text.toString()),
                password = binding.password.text.toString()
            )
        }
        binding.register.setOnClickListener {
            register(
                User(email = binding.username.text.toString()),
                password = binding.password.text.toString()
            )
        }

    }

    private fun isUserAuth() {
        if (viewModel.isUserAuthenticated) {
            navController.navigate(R.id.action_loginFragment_to_nav_home)
        }
    }

    private fun login(user: User, password: String) {
        viewModel.login(user, password).asLiveData().observe(viewLifecycleOwner, {
            if (it == true) {
                navController.navigate(R.id.action_loginFragment_to_nav_home)
            }
            })
    }

    private fun register(user: User, password: String) {
        if (viewModel.register(user, password)) {
            Toast.makeText(context, "Регистрация прошла успешно!", Toast.LENGTH_SHORT)
                .show()
        } else {
            Toast.makeText(context, "Регистрация провалена!", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

}


